### PR TITLE
[WIP][feature] support `create`,`alter`,`refresh`,`drop` stmt syntax for multi table materialized view

### PR DESCRIPTION
**Pull request by yangzhg**
# Proposed changes

Issue Number: #7503 

## Problem Summary:

support `create`,`alter`,`refresh`,`drop` stmt for multi table materialized view

**this pr is contains the syntax support only**

### create

```sql
create materialized view mv_name     
build [immediate | deferred] 		
refresh [fast | complete | never].     
on [commit | demand] 		
start with start_time		
next interval	
PARTITION BY [range|list]   
DISTRIBUTED BY hash(cols..) BUCKETS 16 
PROPERTIES()      
as	           
select ...;    
```

### DESC

```sql
DESCRIBE test_mv_view;
```

###  SHOW CREATE

```sql
SHOW CREATE MATERIALIZED VIEW test_mv_view;
```

### DROP

```sql
DROP MATERIALIZED VIEW test_mv_view;
```

### ALTER

```sql
alter materialized view mv1
refresh [fast | complete | never]
on [commit | demand]
start with start_time
next interval
```

### refresh

```sql
REFRESH MATERIALIZED VIEW test_mv_view [complete];
```

## Checklist(Required)

1. Does it affect the original behavior: (Yes/No/I Don't know)
2. Has unit tests been added: (Yes/No/No Need)
3. Has document been added or modified: (Yes/No/No Need)
4. Does it need to update dependencies: (Yes/No)
5. Are there any changes that cannot be rolled back: (Yes/No)

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: fe/fe-core/src/main/cup/sql_parser.cup [hunk 1]
```diff
@@ -240,25 +240,25 @@ parser code {:
...
-    KW_DATA, KW_DATABASE, KW_DATABASES, KW_DATE, KW_DATETIME, KW_DAY, KW_DECIMAL, KW_DECOMMISSION, KW_DEFAULT, KW_DESC, KW_DESCRIBE,
+    KW_DATA, KW_DATABASE, KW_DATABASES, KW_DATE, KW_DATETIME, KW_DAY, KW_DECIMAL, KW_DECOMMISSION, KW_DEFAULT, KW_DEFERRED, KW_DEMAND, KW_DESC, KW_DESCRIBE,
     KW_DELETE, KW_UPDATE, KW_DIAGNOSE, KW_DISK, KW_DISTINCT, KW_DISTINCTPC, KW_DISTINCTPCSA, KW_DISTRIBUTED, KW_DISTRIBUTION, KW_DYNAMIC, KW_BUCKETS, KW_DIV, KW_DOUBLE, KW_DROP, KW_DROPP, KW_DUPLICATE,
     KW_ELSE, KW_ENABLE, KW_ENCRYPTKEY, KW_ENCRYPTKEYS, KW_END, KW_ENGINE, KW_ENGINES, KW_ENTER, KW_ERRORS, KW_EVENTS, KW_EXCEPT, KW_EXCLUDE,
     KW_EXISTS, KW_EXPORT, KW_EXTENDED, KW_EXTERNAL, KW_EXTRACT,
-    KW_FALSE, KW_FEATURE, KW_FOLLOWER, KW_FOLLOWING, KW_FREE, KW_FROM, KW_FIELDS, KW_FILE, KW_FILTER, KW_FIRST, KW_FLOAT, KW_FOR, KW_FORCE, KW_FORMAT, KW_FRONTEND, KW_FRONTENDS, KW_FULL, KW_FUNCTION, KW_FUNCTIONS,
+    KW_FAST, KW_FALSE, KW_FEATURE, KW_FOLLOWER, KW_FOLLOWING, KW_FREE, KW_FROM, KW_FIELDS, KW_FILE, KW_FILTER, KW_FIRST, KW_FLOAT, KW_FOR, KW_FORCE, KW_FORMAT, KW_FRONTEND, KW_FRONTENDS, KW_FULL, KW_FUNCTION, KW_FUNCTIONS,
```
Comment (morningman) [inline fe/fe-core/src/main/cup/sql_parser.cup:251]: Do we need to add all these new keywords to the `keywords` region?

File: fe/fe-core/src/main/java/org/apache/doris/analysis/AlterMVStmt.java [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+package org.apache.doris.analysis;
+
+import org.apache.doris.catalog.Catalog;
+import org.apache.doris.common.AnalysisException;
+import org.apache.doris.common.ErrorCode;
+import org.apache.doris.common.ErrorReport;
+import org.apache.doris.mysql.privilege.PrivPredicate;
+import org.apache.doris.qe.ConnectContext;
+
+public class AlterMVStmt extends DdlStmt  {
```
Comment (morningman) [inline fe/fe-core/src/main/java/org/apache/doris/analysis/AlterMVStmt.java:27]: Unify to either "MV" or "MaterializedView" for class naming?

Comment (yangzhg) [reply fe/fe-core/src/main/java/org/apache/doris/analysis/AlterMVStmt.java]: done

File: fe/fe-core/src/main/cup/sql_parser.cup (hunk 1 above)
Comment (yangzhg) [reply fe/fe-core/src/main/cup/sql_parser.cup:251]: If you don't add it, the parsing can't use these keywords.

Comment (morningman) [reply fe/fe-core/src/main/cup/sql_parser.cup:251]: No, I means you just add `KW_BUILD` to the `keywords`. Why not adding all other new keywords to the `keywords`?

**Review by morningman — DISMISSED** [review]
LGMT except for some tiny comments.
And need to rebase to resolve the conflict.

File: fe/fe-core/src/main/cup/sql_parser.cup (hunk 1 above)
Comment (yangzhg) [reply fe/fe-core/src/main/cup/sql_parser.cup:251]: all new key words had already in key words

File: fe/fe-core/src/main/java/org/apache/doris/analysis/DropMaterializedViewStmt.java [hunk 3]
```diff
@@ -49,8 +51,18 @@ public DropMaterializedViewStmt(boolean ifExists, String mvName, TableName table
...
+        this.ifExists = ifExists;
+        this.tableName = null;
+    }
+
     public String getMvName() {
-        return mvName;
+        if (mtmvName != null) {
+            return mvName;
+        } else {
+            return mtmvName.toString();
```
Comment (stalary) [inline fe/fe-core/src/main/java/org/apache/doris/analysis/DropMaterializedViewStmt.java:64]: NPE may occur if mtmvName is null.

File: fe/fe-core/src/main/java/org/apache/doris/alter/Alter.java [hunk 4]
```diff
@@ -127,6 +138,10 @@ public void processDropMaterializedView(DropMaterializedViewStmt stmt) throws Dd
         ((MaterializedViewHandler) materializedViewHandler).processDropMaterializedView(stmt, db, olapTable);
     }
 
+    public void processRefreshMaterializedView(RefreshMaterializedViewStmt stmt) throws DdlException {
+        throw new DdlException("DROP MATERIALIZED VIEW is not implemented: " + stmt.toSql());
```
Comment (stalary) [inline fe/fe-core/src/main/java/org/apache/doris/alter/Alter.java:142]:
```suggestion
        throw new DdlException("Refresh materialized view is not implemented: " + stmt.toSql());
```

**Review by morningman — APPROVED** [review]
LGTM

**Pull request merged**